src/Servers/HttpSys/src/UrlPrefixCollection.cs

```diff
@@ -138,10 +142,36 @@ internal void RegisterAllPrefixes(UrlGroup urlGroup)
             {
                 _urlGroup = urlGroup;
                 // go through the uri list and register for each one of them
-                foreach (var pair in _prefixes)
+                // Call ToList to avoid modification when enumerating.
+                foreach (var pair in _prefixes.ToList())
                 {
-                    // We'll get this index back on each request and use it to look up the prefix to calculate PathBase.
-                    _urlGroup.RegisterPrefix(pair.Value.FullPrefix, pair.Key);
+                    var urlPrefix = pair.Value;
+                    if (urlPrefix.PortValue == 0)
+                    {
+                        if (urlPrefix.IsHttps)
+                        {
+                            throw new InvalidOperationException("Cannot bind to port 0 with https.");
+                        }
+
+                        for (var port = BasePort; port < MaxPort; port++)
```

Comment:
Always starting at BasePort means you're always going to hit the same conflicts over and over again. The tests used a static field to keep the last port. Sockets do something similar internally.

Reply:
Yeah, I was going to ask if I should keep that behavior. I'll restore it.

Reply:
How much do we care about looping back to the original port. Ex NextPort is 7999, should we reloop back to 5000?

Reply:
If it's static it must loop at some point or you limit the number of instances you can start in a test process.